dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -241,15 +251,22 @@ def refute_estimate(self):
             else:
                 groups = self.preprocess_data_by_treatment()
                 for key_train, _ in groups:
-                    X_train = groups.get_group(key_train)[self._chosen_variables].values
-                    outcome_train = groups.get_group(key_train)['y'].values
+                    base_train = groups.get_group(key_train).sample(frac=self._test_validation_split.base)
+                    train_set = set( [ tuple(line) for line in base_train.values ] )
```

Comment:
there must be a native way to do this in pandas.---retrieve a dataframe that does not include given indices.

Reply:
There seemed to be no direct ways of doing this. I tried a few methods, out of which the current method seems to work best.  I think the code would be even more efficient, if there is an in built method to find the union of n DataFrames, similar to the concat method.